MicroBenchmarks/LoopVectorization/LoopInterleaving.cpp

```diff
@@ -0,0 +1,313 @@
+// This program tests performance impact of Interleaving Count with varying loop
+// iteration count for different types of loop, such as loops with or
+// without reduction inside it, loops with different vectorization widths.
+#include <iostream>
+#include <memory>
+#include <random>
+
+#include "benchmark/benchmark.h"
+
+#define ELEMENTS 2048
+#define ALIGNED16 __attribute__((aligned(16)))
+
+static std::mt19937 rng;
+unsigned int g_sum = 0;
+
+int A[ELEMENTS] ALIGNED16;
+int B[ELEMENTS] ALIGNED16;
+int C[ELEMENTS] ALIGNED16;
+
+// Initialize arrays with random numbers.
+static void init_data(unsigned N) {
+  std::uniform_int_distribution<int> distrib(std::numeric_limits<int>::min(),
+                                             std::numeric_limits<int>::max());
+  for (unsigned I = 0; I < N; I++) {
+    A[I] = distrib(rng);
+    B[I] = distrib(rng);
+    C[I] = distrib(rng);
+  }
+}
+
+static void __attribute__((noinline)) loopWithVW4IC1(int Iterations) {
+#pragma clang loop vectorize_width(4) interleave(disable)
+  for (int J = 0; J < Iterations; J++) {
+    A[J] = B[J] + C[J];
+  }
+}
+
+static void __attribute__((noinline)) loopWithVW4IC2(int Iterations) {
+#pragma clang loop vectorize_width(4) interleave_count(2)
```

Comment:
Not sure if that's possible, but could the `vectorize_with` and `internalave_count` be template arguments?

Reply:
Since the vectorization hints are added via the `#pragma` directive, the `#` inside the macro function definition is causing problem & I couldn't figure out a way to escape the `#` even with macro expansion.

Reply:
Figured a way to do this using the `_Pragma` operator.